libs/lib-client/define-params/src/params-schema/define-params.service.ts

```diff
@@ -0,0 +1,73 @@
+import { Injectable } from '@angular/core';
+import { AbstractControl, FormArray, FormBuilder } from '@angular/forms';
+import { ValueType } from '@flogo-web/core';
+import { isEmpty } from 'lodash';
+
+@Injectable()
+export class DefineParamsService {
+  constructor(private _fb: FormBuilder) {}
+
+  getInitForm() {
+    return this._fb.group({
+      input: this._fb.array([]),
+      output: this._fb.array([]),
+    });
+  }
+
+  getFormWithData(input, output) {
```

Comment:
1. We're missing the types for input and output
2. Since this is creating a new object maybe the name should be `createFormWithData()` or `makeFormWithData()`. Because of the way we usually use services if I read `getX()` I wouldn't think I'm creating a new instance of something
1. Do we need a separate function for empty form vs form with data? Could we maybe merge it to:
```ts
createForm({ input, output }: { input?, output? } = {}) {
	return this._fb.group({
      input: input ? this.mapAttributesToFormArray(input) : this._fb.array([]),
      output: output ? this.mapAttributesToFormArray(output):  this._fb.array([]),,
    });
}
```